### PR TITLE
refactor: replace dialog

### PR DESCRIPTION
**Pull request by gabitoesmiapodo**
# Description:

- Replaced Dialog component from `db-ui-toolkit` for [Faceless UI's modal](https://facelessui.com/docs/getting-started). It's pretty similar, but it works better, it's better documented, etc. Most importantly: if you have any questions you'll have to ask someone that's not me.
- Added a specific component for the "single token" thing instead of using the dropdown button for multiple purposes. The dropdown button should be a dropdown button and nothing else.
- Fixed a bug where `TokenInput`'s `Textfield` wouldn't pick up some CSS.
- Updated the readme file.
- Updated `db-ui-toolkit`'s version.

# Steps:

There's a `origin/replace-dialog-test` branch were I added a demo `TokenInput` were you can see this working.

Deployment's here https://dappbooster-git-replace-dialog-test-bootnode.vercel.app/

## Type of change:

- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Enhancement
- [x] Refactoring
- [ ] Chore

# How Has This Been Tested?

- [x] Manual testing
- [ ] Automated tests
- [ ] Other (explain)

# Remember to check that:

- Your code follows the style guidelines of this project
- You have performed a self-review of your code
- You have commented your code in hard-to-understand areas
- You have made corresponding changes to the documentation
- Your changes generate no new warnings

# Screenshots

Aug-21-2024 19-03-47

### CODE REVIEW

**Comment by fernandomg** [comment]
> * Added a specific component for the "single token" thing instead of using the dropdown button for multiple purposes. The dropdown button should be a dropdown button and nothing else.

Love this, thanks!

**Comment by fernandomg** [comment]
> There's a `origin/replace-dialog-test` branch were I added a demo `TokenInput` were you can see this working.
> 
> Deployment's here https://dappbooster-git-replace-dialog-test-bootnode.vercel.app/

❤️

File: src/components/sharedComponents/TokenInput/index.tsx [hunk 1]
```diff
@@ -46,12 +48,11 @@ export const CloseButton = styled(BaseCloseButton)`
...
 `
 
-interface TokenInputProps extends Omit<TokenSelectProps, 'onError' | 'onTokenSelect'> {
+interface TokenInputProps extends Omit<TokenSelectProps, 'onError'> {
   singleToken?: boolean
   token?: Token
   onAmountSet: (amount?: string) => void
-  onTokenSelect?: (token: Token | undefined) => void
-  onError?: (error?: string) => void
+  onError: (error?: string) => void
```
Comment (fernandomg) [inline src/components/sharedComponents/TokenInput/index.tsx:55]: making `onError` optional was intentional since #246

**Review by fernandomg — APPROVED** [review]
I love how straightforward the migration was 🥇

All looks and behaves great.

I just want to address that tiny change around `onError` not being optional

File: src/components/sharedComponents/TokenInput/index.tsx (hunk 1 above)
Comment (gabitoesmiapodo) [reply src/components/sharedComponents/TokenInput/index.tsx]: My bad. Fixed.

**Pull request merged**